shell/browser/native_window_views.cc

```diff
@@ -751,17 +751,25 @@ void NativeWindowViews::SetFullScreen(bool fullscreen) {
   else if (fullscreen)
     widget()->native_widget_private()->Show(ui::SHOW_STATE_FULLSCREEN,
                                             gfx::Rect());
-
+#endif
   // Auto-hide menubar when in fullscreen.
   if (fullscreen) {
     menu_bar_visible_before_fullscreen_ = IsMenuBarVisible();
     SetMenuBarVisibility(false);
   } else {
+#if BUILDFLAG(IS_WIN)
+    // No fullscreen -> fullscreen video -> un-fullscreen video results
+    // in `NativeWindowViews::SetFullScreen(false)` being called twice.
+    // `menu_bar_visible_before_fullscreen_` is always false on the
+    //  second call which results in `SetMenuBarVisibility(false)` no
+    // matter what. We check `leaving_fullscreen` to avoid this.
+    if (!leaving_fullscreen)
+      return;
+#endif
```

Comment:
Can we restore the two `#ifdef` cases to make this easier to read? I think it's ok if we need to copy some logic here.

Reply:
Is this ok? [`2558922` (#43402)](https://github.com/electron/electron/pull/43402/commits/2558922a2b9b86396e928846848586fe4a2a3285)